### DISCUSSION TITLE
Merging routes from multiple directories in the Vite build, results in duplicate routes in devtools/`router.getRoutes()`, understanding normalization

### DISCUSSION THREAD

**Discussion by josh-hemphill** [vuejs/router#2363] [Help and Questions] [answered]
I did my best to match up all the route paths and names before passing them into the vue router, and they seem to be fine there, but once I load the router and try to pull the route records with `router.getRoutes()` a bunch of the subdirectories have duplicate base-level entries, which messes things up if I want to drive a Nav UI from the results of it.
Not sure what I'm missing or if I'm misunderstanding the normalization steps the router takes internally, but it also effects what routes are listed in the devtools which can get pretty confusing.
Should I be trying to use `alias`es where multiple routes result in the same page instead of separate routes? Maybe that's effecting it?

**Comment by posva** [comment, marked as answer]
yes, you should use alias but it also works without it. Impossible to say without the actual config or a reproduction though
